rs/execution_environment/src/execution/replicated_query.rs

```diff
@@ -127,14 +127,28 @@ pub fn execute_replicated_query(
     let memory_usage = canister.memory_usage();
     let message_memory_usage = canister.message_memory_usage();
 
-    let api_type = ApiType::replicated_query(time, req.method_payload().to_vec(), *req.sender());
+    let request_metadata = match &req {
+        CanisterCall::Request(request) => request.metadata.for_downstream_call(),
+        _ => RequestMetadata::for_new_call_tree(time),
```

Comment:
Can this catch-all ever occur? I don't think so since we shouldn't be ever running a response as a replicated query.

Reply:
`CanisterCall` is actually either a `Request` or `Ingress`, never a `Response`. (execution environment nomenclature to capture "outside" inputs to the canister). Let me list out the two cases explicitly, the catch all is misleading here.